arch/arm/src/stm32/Kconfig

```
@@ -1129,15 +1129,80 @@ config ARCH_CHIP_STM32F469N
 	select STM32_STM32F469
 	select STM32_HAVE_ETHMAC
 
+config ARCH_CHIP_STM32G474C
```

It seems to me that this is not correct. You should use chip names from [here](https://www.st.com/content/ccc/fragment/product_related/line_information/line_level_diagram/group0/cc/a2/82/c1/1b/2a/45/35/stm32g4x4_line_ln2130/files/stm32g4x4_line_ln2130.jpg/_jcr_content/translations/en.stm32g4x4_line_ln2130.jpg) (eg. STM32G474**CE**, STM32G474**CC** ... not  STM32G474**C**)
Each chip version should be a single entry in the "STM32 Chip Selection" choice menu.
Also I don't think that we need a distinction between LQFP/UFQFPN and LQFP/BGA packages. 
This does'nt affect OS in any way

---

I'm working on this change...

---

If I do what you suggest, the second letter determines the flash size:

STM32G474x**E** = 512Kbyte
STM32G474x**C** = 256Kbyte
STM32G474x**B** = 128Kbyte

How do I handle that? Like this:

```suggestion
config ARCH_CHIP_STM32G474CB
	bool "STM32G474CB"
	select STM32_STM32G47XX
	select STM32_STM32G474C
	select STM32_FLASH_CONFIG_B
```

Note that STM32_FLASH_CONFIG_B is part of a 'choice' and I don't see any other chip doing it that way.

---

> Also I don't think that we need a distinction between LQFP/UFQFPN and LQFP/BGA packages.
> This does'nt affect OS in any way

I removed distinction between LQFP/UFQFPN and LQFP/BGA packages in cc206e81b3b3d920c8031c025cbf6a368325febc.

---

Looks good. 
But I'm not sure if we need additional options for STM32_STM32G474M, STM32_STM32G474R, etc.
Shouldn't we keep chip-specific options just under ARCH_CHIP_STM32G474M, ARCH_CHIP_STM32G474R, etc ?
Then leave only STM32_STM32G47XX as the common configuration for the G47XX familly.
There is currently no consistency regarding this in Kconfig

---

> Looks good.
> But I'm not sure if we need additional options for STM32_STM32G474M, STM32_STM32G474R, etc.
> Shouldn't we keep chip-specific options just under ARCH_CHIP_STM32G474M, ARCH_CHIP_STM32G474R, etc ?
> Then leave only STM32_STM32G47XX as the common configuration for the G47XX familly.
> There is currently no consistency regarding this in Kconfig

Yes, Kconfig is inconsistent. I tried to follow what most other chips were doing.

What would your preference be?

---

I don't have any strong opinion about it, so let's leave it as it is.

---

OK